plugins/plugin-maven/che-plugin-maven-ide/src/main/java/org/eclipse/che/plugin/maven/client/command/MavenCommandModel.java

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.maven.client.command;
+
+import org.eclipse.che.ide.CommandLine;
+
+/**
+ * Model of the Maven command line.
+ *
+ * @author Artem Zatsarynnyi
+ */
+class MavenCommandModel {
+
+    private String workingDirectory;
+    private String arguments;
+
+    MavenCommandModel(String workingDirectory, String arguments) {
+        this.workingDirectory = workingDirectory;
+        this.arguments = arguments;
+    }
+
+    /** Crates {@link MavenCommandModel} instance from the given command line. */
+    static MavenCommandModel fromCommandLine(String commandLine) {
+        final CommandLine cmd = new CommandLine(commandLine);
+
+        String workingDirectory = null;
+
+        if (cmd.hasArgument("-f")) {
```

Comment:
Good idea for the further improvements of CommandLine class. :+1: 
I think it should be done in a separate task since I've just moved this code 'as-is' from another class during refactoring.